### PR TITLE
Vendor shutil.get_terminal_size.

### LINKED ISSUE #9815: ImportError: No module named shutil_get_terminal_size
Update from @carreau : 
 Reopening, tagging 5.4 we should vendor shutil_get_terminal_size.

--- 
After installing ipython `sudo apt-get install ipython-notebook` , `ipython` it appears a error, as follows:

jiangyuping@Lenovo:~$ ipython
Traceback (most recent call last):
  File "/usr/local/bin/ipython", line 4, in <module>
    from IPython import start_ipython
  File "/usr/local/lib/python2.7/dist-packages/IPython/**init**.py", line 48, in <module>
    from .core.application import Application
  File "/usr/local/lib/python2.7/dist-packages/IPython/core/application.py", line 25, in <module>
    from IPython.core import release, crashhandler
  File "/usr/local/lib/python2.7/dist-packages/IPython/core/crashhandler.py", line 28, in <module>
    from IPython.core import ultratb
  File "/usr/local/lib/python2.7/dist-packages/IPython/core/ultratb.py", line 128, in <module>
    from IPython.utils.terminal import get_terminal_size
  File "/usr/local/lib/python2.7/dist-packages/IPython/utils/terminal.py", line 22, in <module>
    from backports.shutil_get_terminal_size import get_terminal_size as _get_terminal_size
ImportError: No module named shutil_get_terminal_size

### PR DESCRIPTION
**Pull request by Carreau**
Use the vendored version only if import error.
That should fix import error for many users.

Closes #9815.

This is only against the 5.x branch as this is in the stdlib on Python 3

### CODE REVIEW

File: IPython/utils/terminal.py [hunk 1]
```diff
@@ -19,7 +19,10 @@
     from shutil import get_terminal_size as _get_terminal_size
 except ImportError:
     # use backport on Python 2
-    from backports.shutil_get_terminal_size import get_terminal_size as _get_terminal_size
+    try:
+        from backports.shutil_get_terminal_size import get_terminal_size as _get_terminal_size
+    except ImportError:
+        from _get_terminal_size import _get_terminal_size
```
Comment (takluyver) [inline IPython/utils/terminal.py]: Let's put the `.` in for an explicit relative import. It's clearer, and if someone adds `from __future__ import absolute_import`, it won't break.

Comment (Carreau) [inline IPython/utils/terminal.py]: good point.

File: IPython/utils/_get_terminal_size.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# vendored version of backports.get_terminal_size as nemesapece package are a
+# mess and break, especially on ubuntu. This file is under MIT Licence.
```
Comment (takluyver) [inline IPython/utils/_get_terminal_size.py]: It should therefore include the copyright notice attributing the original file. And, technically, the permissions part of the MIT license as well, although I'm not going to complain if we just say 'MIT licence'.

The relevant bit of the MIT licence says:

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

**Comment by Carreau** [comment]
Licence added, link to repo, commit sha and date, updated to master version

**Pull request merged**